src/spec.rs

```diff
@@ -84,9 +88,19 @@ impl FromStr for Endpoints {
             .map(|addr| {
                 let span = tracing::error_span!("parse_addr", ?addr);
                 let _g = span.enter();
-                match addr.parse() {
-                    Ok(ep) => Ok(ep),
-                    Err(_) => parse_error!("invalid socket address"),
+
+                let mut parts = addr.splitn(2, '#');
+                match (parts.next(), parts.next()) {
+                    (Some(addr), h2) => match addr.parse() {
+                        Ok(addr) => Ok((
+                            addr,
+                            EndpointMeta {
+                                h2: h2.map(|proto| proto == "h2").unwrap_or(false),
```

Comment:
nit: I think this is the same as 
```suggestion
                                h2: h2.contains("h2"),
```

Reply:
```
🔬 This is a nightly-only experimental API. (option_result_contains #62358)
```

Reply:
Can also just do `unwrap_or_default` for now